src/Frontend/Modules/Mailmotor/Actions/Subscribe.php

```diff
@@ -61,6 +62,20 @@ public function execute(): void
             );
 
             $doubleOptIn = false;
+        } catch (Exception $exception) {
+            $reason = json_decode($exception->getMessage());
+            // check if the error is one from mailchimp
+            if ($reason === false) {
+                throw $exception;
+            }
+
+            $this->template->assign('mailmotorSubscribeHasFormError', true);
```

Comment:
Wouldn't it be better to have a separate error in the template for Mailchimp specific errors? And also passing the reason to it?

Reply:
We could, but this would work in all custom themes as well without modification.
Apart from that, I think it is a little bit dangerous to just parse the errors from MailChimp since I just get a bad request back, but you never know what info will be included. On top of that are the returned errors in English, so on a site in a different language you would get multiple languages on a page with no proper way of translating it

Reply:
maybe log the error into a error log, because with your fix you will never know what the error was.